lib/wrappers/transform-node.js

```diff
@@ -13,6 +14,34 @@ module.exports = class TransformNodeWrapper extends NodeWrapper {
       cachePath: this.cachePath,
     });
     this.callbackObject = this.nodeInfo.getCallbackObject();
+    this.callbackObject.revised = () => {
+      this.revised();
+    };
+
+    // This weakmap holds references from inputNode --> last known revision #
+    // If the any inputNode's ref does not match what is stored in here then we
+    // know a modification has happened so we call the build method
+    this.inputRevisions = new WeakMap();
+  }
+
+  shouldBuild() {
+    let shouldBuild = false;
+
+    // The plugin has no input nodes so it's build method should not
+    // be called after the first build
+    if (this.inputNodeWrappers.length === 0 && this.revision > 0) {
+      return false;
+    }
+
+    this.inputNodeWrappers.forEach(wrapper => {
+      if (this.inputRevisions.get(wrapper) !== wrapper.revision) {
+        logger.debug(`${wrapper.toString()} has been revised since last build.`);
```

Comment:
Once you try this out in an app, let's insure this debug logging is as actionable as possible when "rebuild" performance problem arises.

Reply:
Working on this now as I run it in real apps